src/main/java/io/skai/okta/internshipstreamingserviceparserxml/services/converter/EpisodeConverter.java

```diff
@@ -23,18 +32,47 @@ public Episode map(LostFilmRssItem lostFilmRssItem) {
                       .build();
     }
 
-    private String extract(String from, String value) {
-        final String[] result = from.replaceAll("\\(", ")").split("\\)");
-        switch (value) {
-            case "tvsTitle":
-                return result[1];
-            case "season":
-                return result[3].substring(1, 3);
-            case "episode":
-                return result[3].substring(4, 6);
+    private String extract(String from, EpisodeField field) {
+        switch (field) {
+            case TV_SERIES_TITLE:
+                return extractTvsTitle(from);
+            case SEASON_NUMBER:
+                return extractSeason(from);
+            case EPISODE_NUMBER:
+                return extractEpisode(from);
             default:
-                throw new IllegalStateException("Unexpected value: " + value);
+                throw new IllegalStateException("Unexpected field: " + field);
+        }
+    }
+
+    private String extractTvsTitle(String from) {
+        Matcher matcher = TVS_TITLE_PATTERN.matcher(from);
+        if (matcher.find()) {
+            return matcher.group(0);
+        }
+        throw new InvalidParameterException("Format of \"title\" string is illegal");
+    }
+
+    private String extractSeason(String from) {
+        Matcher matcher = SEASON_PATTERN.matcher(from);
+        if (matcher.find()) {
+            return matcher.group(0);
         }
+        throw new InvalidParameterException("Format of \"title\" string is illegal");
+    }
+
+    private String extractEpisode(String from) {
+        Matcher matcher = EPISODE_PATTERN.matcher(from);
+        if (matcher.find()) {
+            return matcher.group(0);
+        }
+        throw new InvalidParameterException("Format of \"title\" string is illegal");
+    }
```

Comment:
all those 3 methods does the same thing, the only difference which pattern to use, so why we can't pass it as argument and have 1 method instead of 3?

Reply:
omg, you're write :)